WSLBlobNFS/WSLBlobNFS.psm1

```diff
@@ -6,178 +11,539 @@ Set-PSDebug -Strict
+# Internal functions
+#
+function Get-ModuleVersion
 {
+    [CmdletBinding()]
+    param()
+
+    # In local dev env, the module is not installed from gallery. Hence, Get-InstalledModule will fail.
+    # Hence, suppress the error and return 0.0.0 as the module version.
+    $blobNFSModule = Get-InstalledModule -Name $moduleName 2>$null
+    if ($blobNFSModule -eq $null)
+    {
+        return "0.0.0"
+    }
+
+    return $blobNFSModule.Version.ToString()
+}
+
+function Write-Success
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $true)][string]$message
+    )
+
+    Write-Host $message -ForegroundColor DarkGreen
+}
+
+function Execute-WSL
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $true)][string]$wslcommand
+    )
+
+    Write-Debug "Executing $wslcommand"
+    wsl -d $distroName -u $userName -e bash -c $wslcommand
+}
+
+function Copy-WSLFiles
+{
+    [CmdletBinding()]
+    param()
+
+    # Note: Quote the path with '' to preserve space
+    Write-Verbose "Copying necessary linux files from $modulePathForLinux"
+
+    Execute-WSL "mkdir -p $linuxScriptsPath"
+
+    Execute-WSL "cp '$modulePathForLinux/$wslScriptName' $linuxScriptsPath/$wslScriptName"
+
+    # Files saved from windows will have \r\n line endings. Hence, we need to remove \r.
+    Execute-WSL "sed -i -e 's/\r$//' $linuxScriptsPath/$wslScriptName"
+
+    Execute-WSL "chmod +x $linuxScriptsPath/$wslScriptName"
+
+    Execute-WSL "cp '$modulePathForLinux/$queryScriptName' $linuxScriptsPath/$queryScriptName"
+
+    # Files saved from windows will have \r\n line endings. Hence, we need to remove \r.
+    Execute-WSL "sed -i -e 's/\r$//' $linuxScriptsPath/$queryScriptName"
+
+    Execute-WSL "chmod +x $linuxScriptsPath/$queryScriptName"
+
+    Execute-WSL "ls -l $linuxScriptsPath | grep -E '$wslScriptName|$queryScriptName' > /dev/null 2>&1"
+}
+
+function Intall-WSLBlobNFS-Internal
+{
+    [CmdletBinding()]
+    param()
+
     # Check if wsl is installed or not.
-    $wslstatus = wsl -l -v
-    if($LASTEXITCODE -eq 1)
+    wsl -l -v > $null 2>&1
+    $wslstatus = $LASTEXITCODE
+
+    # Check if the distro is installed or not.
+    # Note: executing a command such as ls is required else if the distro is installed, then control will not come out of wsl command.
+    wsl -d $distroName ls > $null 2>&1
+    $distrostatus = $LASTEXITCODE
+
+    if($wslstatus -eq 1)
     {
         Write-Host "WSL is not installed. Installing WSL..."
 
-        wsl --install -d Ubuntu-22.04
-        Write-Host "Restart the VM to complete WSL installation, and then Run the script again with Initialize-WSLBlobNFS to continue the WSL setup."
+        wsl --install -d $distroName
+        # To-do: Check if the distro is installed successfully or not.
+
+        # To-do: Prompt user to restart the machine to complete WSL installation.
+        Write-Success "Restart the VM to complete WSL installation, and then run Initialize-WSLBlobNFS to continue the WSL setup."
```

Comment:
VM -> machine

also before this line, following would be more comforting to the user

Write-Success "WSL distro $distroName successfully installed!"